src/crimson/os/seastore/journal/circular_bounded_journal.h

```diff
@@ -30,7 +30,27 @@ using NVMeBlockDevice = nvme_device::NVMeBlockDevice;
 /**
  * CircularBoundedJournal
  *
- * TODO: move record from CircularBoundedJournal to RandomBlockManager
+ * 
+ * CircularBoundedJournal (CBJournal) is the journal that works like circular
+ * queue. With CBJournal, Seastore will append some of the records if the size
+ * of the record is small (most likely metadata), at which point the head
+ * (written_to) will be moved. Then, eventually, Seastore applies the records
+ * in CBjournal to RBM---the tail (applied_to) is advanced (TODO).
+ *
+ * - Commit time
+ * After submit_record is done, last_committed_record_base and written_to are
+ * advanced---last_committed_record_base indicates the last committed record and
+ * written_to represents where the new record will be appended. Note that
+ * applied_to is not changed here.
+ *
+ * - Replay time
+ * At replay time, CBJournal begins to replay records in CBjournal by reading
+ * records from applied_to. Then, CBJournal examines whether the records is valid
+ * one by one, at which point last_committed_record_base and written_to are
+ * recovered if the valid record is founded. Note that only applied_to is managed
```

Comment:
What makes a record "valid"?

Reply:
there are two conditions: 1) decoding without an exception, 2) correct sequence number (this should be less than prior valid record).